builder/builder/builder.go

```diff
@@ -570,6 +572,40 @@ func (b *Builder) OnPayloadAttribute(attrs *types.BuilderPayloadAttributes) erro
 		return fmt.Errorf("could not get validator while submitting block for slot %d - %w", attrs.Slot, err)
 	}
 
+	proposerPubkey, err := utils.HexToPubkey(string(vd.Pubkey))
+	if err != nil {
+		return fmt.Errorf("could not parse pubkey (%s) - %w", vd.Pubkey, err)
+	}
+
+	// BOLT: by default, the proposer key is the constraint signer key
+	pubkey, err := utils.HexToPubkey(string(vd.Pubkey))
+	if err != nil {
+		log.Error("could not parse pubkey", "pubkey", vd.Pubkey, "err", err)
+	}
+	constraintsPubkeys := []phase0.BLSPubKey{pubkey}
+
+	// BOLT: get delegations for the slot
+	delegations, err := b.relay.GetDelegationsForSlot(attrs.Slot)
+	if err != nil {
+		log.Error("could not get delegations for slot, using default validator key", "slot", attrs.Slot, "err", err)
+	}
+
+	if len(delegations) > 0 {
+		// If there are delegations, reset the constraintsPubkeys (i.e. remove the validator key as authorized)
```

Comment:
So when delegating to a key, the proposer key is revoked? If so, why? Is this outlined in the API specs too?

Reply:
Don't see a clear reason to either remove the proposer key or keep it in the list, but since it's a list I suppose we can keep it. To me it seemed more logical that if a proposer decides to delegate to another key, its own key shouldn't be used anymore for signing constraints. Wdyt?

Reply:
I agree, it's fine as it is then. I still suggest documenting this behaviour in the API specs / docs